ipl-tournament/src/com/tarktech/training/ipl/SimpleMatchScheduler.java

```diff
@@ -13,17 +13,41 @@
 
         LocalDate matchScheduleDate = startDate;
 
-        for (int i = 0; i < teams.size(); ++i) {
-            for (int j = i + 1; j < teams.size(); ++j) {
-                CricketMatch cricketMatch = new CricketMatch(teams.get(i), teams.get(j));
-                cricketMatch.scheduleOn(matchScheduleDate);
-
-                scheduledMatches.add(cricketMatch);
-
-                matchScheduleDate = matchScheduleDate.plusDays(1);
+        ArrayList<String> schedule = new ArrayList<>();
+        int r = 0;
+        int i = 0;
+        int j = teams.size()-1;
+
+        if (teams.size() % 2 == 0) {
+            while (r < teams.size() - 1) {
+                while (i < j) {
+                    schedule.add(teams.get(i) + " " + teams.get(j));
+
+                    CricketMatch cricketMatch = new CricketMatch(teams.get(i), teams.get(j));
+                    cricketMatch.scheduleOn(matchScheduleDate);
+                    scheduledMatches.add(cricketMatch);
+                    matchScheduleDate = matchScheduleDate.plusDays(1);
+
+                    i++;
+                    j--;
+                }
+                i = 0;
+                j = teams.size() - 1;
+
+                //right shifting of teams...
+                List<Team> temp = new ArrayList<>(teams);
```

Comment:
We can further simplify the array rotation, using the methods supported by List/ArrayList. This will eliminate the `for loop` completely.

You may want to look at below documentation and see, which methods could be used to achieve this.
https://docs.oracle.com/javase/8/docs/api/java/util/List.html